lib/travis/build/git/netrc.rb

```diff
@@ -3,27 +3,29 @@ module Build
     class Git
       class Netrc < Struct.new(:sh, :data)
         def apply
-          sh.fold 'git.netrc' do
-            sh.echo "Using ${TRAVIS_HOME}/.netrc to clone repository.", ansi: :yellow
-            sh.newline
-            sh.raw "echo -e \"#{netrc}\" > ${TRAVIS_HOME}/.netrc"
-            sh.raw "chmod 0600 ${TRAVIS_HOME}/.netrc"
-          end
+          sh.echo "Using ${TRAVIS_HOME}/#{netrc_filename} to clone repository.", ansi: :yellow
+          sh.newline
+          sh.raw "echo -e #{Shellwords.escape netrc_content} > ${TRAVIS_HOME}/#{netrc_filename}"
+          sh.raw "chmod 0600 ${TRAVIS_HOME}/#{netrc_filename}"
         end
 
         def delete
-          sh.raw "rm -f ${TRAVIS_HOME}/.netrc"
+          sh.raw "rm -f ${TRAVIS_HOME}/#{netrc_filename}"
         end
 
         private
 
-          def netrc
+          def netrc_content
             if data.installation?
               "machine #{data.source_host}\\n  login travis-ci\\n  password #{data.token}\\n"
             else
               "machine #{data.source_host}\\n  login #{data.token}\\n"
```

Comment:
@BanzaiMan sorry I should have drawn attention to this, too, with the `Shellwords.escape` suggestion.  These `\\n` will need to become `\n`.